app/models/classification.rb

```diff
@@ -3,27 +3,29 @@ class Classification < ApplicationRecord
 
   belongs_to :tag
 
+  virtual_column :name, :type => :string
+
   before_save    :save_tag
   before_destroy :delete_tags_and_entries
-  validate       :validate_format_of_name, :validate_uniqueness_on_tag_name
-
-  validates_uniqueness_of :description, :scope => [:parent_id],
-                                        :if    => proc { |c|
-                                          cond = ["parent_id = ? AND description = ?", c.parent_id, c.description]
-                                          unless c.new_record?
-                                            cond.first << " AND id != ?"
-                                            cond << c.id
-                                          end
-                                          c.class.in_region(region_id).exists?(cond)
-                                        }
-  validates_presence_of :name, :description
-  validates_length_of :name, :maximum => 50, :message => "must not exceed 50 characters"
-  validates_length_of :description, :maximum => 255, :message => "must not exceed 255 characters"
-  validates_inclusion_of :syntax,
-                         :in      => %w( string integer boolean ),
-                         :message => "should be one of 'string', 'integer' or 'boolean'"
 
-  virtual_column :name, :type => :string
+  validates :description, :presence => true, :length => {:maximum => 255}
+  validates :description, :uniqueness => {:scope => [:parent_id]}, :if => proc { |c|
+    cond = ["parent_id = ? AND description = ?", c.parent_id, c.description]
+    unless c.new_record?
+      cond.first << " AND id != ?"
+      cond << c.id
+    end
+    c.class.in_region(region_id).exists?(cond)
+  }
```

Comment:
You can use active record for this:

```ruby
scope = c.class.in_region(region_id).where(:parent_id => c.parent_id, :description => :description)
scope = scope.where.not(:id => c.id) unless c.new_record?
c.exists?
```

But that seems odd. We basically wrote our own uniqueness query. Why have us call that just to have them basically call the same query?

What is the point here? Make it unique for a given region?